Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -354,6 +516,35 @@ bool isFunctionJson(string fullName)
             }
         }
 
+        private async Task LocalAppFetch(IRepository repository, DeploymentInfoBase deploymentInfo, string targetBranch, ILogger logger, ITracer tracer)
+        {
+            var zipDeploymentInfo = (ZipDeploymentInfo)deploymentInfo;
+            var content = !string.IsNullOrEmpty(zipDeploymentInfo.ZipURL)
+                ? await DeploymentHelper.GetZipContentFromURL(zipDeploymentInfo, tracer)
+                : Request.Content;
+            //var content = Request.Content;
+            //var fileName = Path.ChangeExtension("app", deploymentInfo.FileExtention);
+            var tempDirPath = repository.RepositoryPath;
+            var targetInfo = FileSystemHelpers.DirectoryInfoFromDirectoryName(tempDirPath);
+            if (targetInfo.Exists)
+            {
+                var moveTarget = Path.Combine(targetInfo.Parent.FullName, Path.GetRandomFileName());
+                using (tracer.Step(string.Format("Renaming extractTargetDirectory({0}) to tempDirectory({1})", targetInfo.FullName, moveTarget)))
+                {
+                    targetInfo.MoveTo(moveTarget);
+                }
+            }          
+            var tempFilePath = Path.Combine(tempDirPath, deploymentInfo.FileName);
+            Directory.CreateDirectory(tempDirPath);
+            using (tracer.Step("Saving request content to {0}", tempFilePath))
+            {
+                await content.CopyToAsync(tempFilePath, tracer);
```

Comment:
>await content.CopyToAsync(tempFilePath, tracer); [](start = 16, length = 48)

So app.war will be copied twice, once to a temp location and then to final destination. What are the pros and cons of this approach?

Reply:
Performance is definitely the biggest con of this approach. I primarily chose this approach as I did not think it would have been feasible to completely redesign the deployment pipeline in the time that I had. I am sure there are other pros and cons and I would love to discuss them if you have time.